moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -132,61 +117,142 @@ bool RuckigSmoothing::applySmoothing(robot_trajectory::RobotTrajectory& trajecto
   }
 
   ruckig::Result ruckig_result;
-  for (size_t waypoint_idx = 0; waypoint_idx < num_waypoints - 1; ++waypoint_idx)
+  bool smoothing_complete = false;
+  robot_trajectory::RobotTrajectory original_trajectory_copy(trajectory);
```

Comment:
Seems to be working just fine without this... will skip this suggestion unless there's a specific reason I should do it.

Reply:
Without deepcopy, the waypoints are not really copied since the pointers still reference the same data ([shallow copy](https://github.com/ros-planning/moveit2/blob/main/moveit_core/robot_trajectory/src/robot_trajectory.cpp#L63)). Ruckig is modifying the waypoint data in each cycle while `original_trajectory_copy` and `trajectory` share the same waypoint data. Both trajectories are affected the same way (besides insertions/deletions). In line 187 you are not resetting the trajectory to the original state from line 121. If I'm not missing something, copying back does nothing since the size and order of the `trajectory` waypoints wasn't changed at all.

Reply:
Done. Thanks